pkg/controller/perconaservermongodbbackup/backaphandler.go

```diff
@@ -0,0 +1,150 @@
+package perconaservermongodbbackup
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	pbapi "github.com/percona/percona-backup-mongodb/proto/api"
+	psmdbv1alpha1 "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1alpha1"
+	"github.com/percona/percona-server-mongodb-operator/pkg/psmdb/backup"
+	"google.golang.org/grpc"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	statusRequested = "requested"
+	statusRejected  = "regected"
+	statusReady     = "ready"
+)
+
+// newBackup creates new Backup
+func newBackupHandler(cr *psmdbv1alpha1.PerconaServerMongoDBBackup) (BackupHandler, error) {
+	if len(cr.Status.Name) == 0 {
+		cr.Status = psmdbv1alpha1.PerconaServerMongoDBBackupStatus{
+			Name:        cr.Name,
+			StorageName: cr.Spec.StorageName,
+			StartAt:     &metav1.Time{},
+			CompletedAt: &metav1.Time{},
+		}
+	}
+	b := BackupHandler{}
+
+	grpcOps := grpc.WithInsecure()
+	conn, err := grpc.Dial(cr.Spec.PSMDBCluster+backup.GetCoordinatorSuffix()+":10001", grpcOps)
+	if err != nil {
+		return b, err
+	}
+
+	client := pbapi.NewApiClient(conn)
+
+	b = BackupHandler{
+		Client: client,
+	}
+
+	return b, nil
+}
+
+// BackupHandler is for working with backup coordinator
+type BackupHandler struct {
+	Client pbapi.ApiClient
+}
+
+// CheckAndUpdateBackup is for check if backup exist and update CR status if true.
+func (b *BackupHandler) CheckAndUpdateBackup(cr *psmdbv1alpha1.PerconaServerMongoDBBackup) (bool, error) {
+	stream, err := b.Client.BackupsMetadata(context.TODO(), &pbapi.BackupsMetadataParams{})
+	if err != nil {
+		return false, err
+	}
+	for {
+		msg, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return false, err
+		}
+		if msg.Metadata.Description == cr.Name {
+			cr.Status.StartAt = &metav1.Time{
+				Time: time.Unix(msg.Metadata.StartTs, 0),
+			}
+			if msg.Metadata.StartTs > 0 {
+				cr.Status.State = statusRequested
+			}
+			if msg.Metadata.EndTs > 0 {
+				cr.Status.CompletedAt = &metav1.Time{
+					Time: time.Unix(msg.Metadata.EndTs, 0),
+				}
+				cr.Status.State = statusReady
+			}
+			for k, v := range msg.Metadata.Replicasets {
+				if len(v.DbBackupName) > 0 {
+					jsonName := strings.Split(v.DbBackupName, "_"+k)
+					cr.Status.Destination = jsonName[0] + ".json"
+					break
+				}
+			}
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
```

Comment:
Still, I'd like to split this method by the logic of "get backup" and "setup status". It'll reduce the cyclomatic complexity of the code as well as the readability in general.
Let "get backup" return `*pbapi.BackupMetadata` and `error`. And if `*pbapi.BackupMetadata` is `nil` so there is no backup